### PR TITLE
Docs/update concepts

### LINKED ISSUE #1870: Update the Concepts section with webpack v4 configuration
In the Concepts section we still have old config snippets and we also are not reflecting the fact that webopack is now cero config out of the box.

Some content might also need to be change so it's compatible with the new webpack default behavior.

### PR DESCRIPTION
**Pull request by EugeneHlushko**
Fixes #1870

### CODE REVIEW

**Comment by EugeneHlushko** [comment]
@jeremenichelli i am not entirely sure that covers it or if format is correct, please have a look

**Comment by TheDutchCoder** [comment]
Hi @EugeneHlushko thanks for this contribution!

I will PR this against the `next` branch, but I do have some feedback, so I'll assign myself as the reviewer so we can work on this!

File: src/content/concepts/index.md [hunk 1]
```diff
@@ -33,6 +34,8 @@ Every dependency is then processed and outputted into files called *bundles*, wh
 
 You can specify an entry point (or multiple entry points) by configuring the `entry` property in the [webpack configuration](/configuration).
 
+T> `entry` defaults to `./src`
```
Comment (TheDutchCoder) [inline src/content/concepts/index.md]: I don't think this needs to be a tip, you can just add it to the introductory paragraph and mention that `./src` is the default value for the `entry` property.

File: src/content/concepts/index.md [hunk 2]
```diff
@@ -50,6 +53,8 @@ T> You can configure the `entry` property in various ways depending the needs of
 
 The **output** property tells webpack where to emit the *bundles* it creates and how to name these files. You can configure this part of the process by specifying an `output` field in your configuration:
 
+T> `output.path` defaults to `./dist`
```
Comment (TheDutchCoder) [inline src/content/concepts/index.md]: Same for this one, just mention it in the intro paragraph.

File: src/content/concepts/index.md [hunk 3]
```diff
@@ -154,3 +151,16 @@ There are many plugins that webpack provides out of the box! Check out our [list
 Using plugins in your webpack config is straightforward - however, there are many use cases that are worth further exploration.
 
 [Learn more!](/concepts/plugins)
+
+
+## Mode
+
+By choosing between `development` and `production` **mode** you can enable webpack's automatic optimizations according to selected option.
```
Comment (TheDutchCoder) [inline src/content/concepts/index.md]: I would maybe change this to: "By setting the `mode` parameter to either `development` or `production`, you can enable webpack's built-in optimizations that correspond with the selected mode."

File: src/content/concepts/index.md [hunk 4]
```diff
@@ -154,3 +151,16 @@ There are many plugins that webpack provides out of the box! Check out our [list
...
+
+```javascript
+module.exports = {
+  mode: 'production'
+};
+```
+
+[Learn more!](/concepts/mode)
```
Comment (TheDutchCoder) [inline src/content/concepts/index.md]: I don't think this page exists (yet), so better take this out for now :)

Comment (EugeneHlushko) [reply src/content/concepts/index.md]: But i did add it within the pr

**Comment by EugeneHlushko** [comment]
@TheDutchCoder i did rebase on `next` for easy reviewing the diff, also fixed 1 lint error. The concepts/mode page was added within this PR, did you review it or missed after changing the target branch?

File: src/content/concepts/index.md (hunk 4 above)
Comment (TheDutchCoder) [reply src/content/concepts/index.md]: Oh yeah, completely missed that because of the branch switch, sorry about that!

**Comment by EugeneHlushko** [comment]
that was fast 🔥🔥🔥

**Comment by TheDutchCoder** [comment]
That's how we roll! 😉 
Great work man, thanks a bunch!

File: src/content/concepts/index.md [hunk 5]
```diff
@@ -48,7 +49,7 @@ T> You can configure the `entry` property in various ways depending the needs of
 
 ## Output
 
-The **output** property tells webpack where to emit the *bundles* it creates and how to name these files. You can configure this part of the process by specifying an `output` field in your configuration:
+The **output** property tells webpack where to emit the *bundles* it creates and how to name these files and defaults to `./dist`. You can configure this part of the process by specifying an `output` field in your configuration:
```
Comment (montogeek) [inline src/content/concepts/index.md]: I would change it to: '...files, it defaults to `./dist`'

File: src/content/concepts/mode.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+Just provide the `mode` option in the config:
+
+```javascript
+module.exports = {
+  mode: 'production'
+};
+```
+
+ 
+or pass it as a cli argument:
```
Comment (montogeek) [inline src/content/concepts/mode.md]: CLI, uppercase

File: src/content/concepts/mode.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+```bash
+webpack --mode=production
+```
+
+The following string values are supported:
```
Comment (montogeek) [inline src/content/concepts/mode.md]: We should add a note or something clarifying that setting `NODE_ENV` doesn't set `mode` automatically, some users are confused by this: https://github.com/webpack/webpack/issues/2537#issuecomment-370384223

**Comment by montogeek** [comment]
It will create a PR with my comments, miss this was merged already

**Comment by EugeneHlushko** [comment]
thank you very much, comments look very valid 👍

**Pull request merged**